swift/Tailscale/LogSink.swift

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+/// A generic interface for sinking log messages from the Swift wrapper
+/// and go
+public protocol LogSink: Sendable {
+    /// An optional file handle.  The go backend will write all internal logs
+    /// to this.  STDOUT_FILENO or a handle to a writable file.
+    var logFileHandle: Int32? { get }
+
+    /// Called for swfit interal logs.
+    func log(_ message: String)
+}
+
+/// Dumps all internal logs to NSLog and go logs to stdout
+struct DefaultLogger: LogSink {
+    var logFileHandle: Int32? = STDOUT_FILENO
+
+    func log(_ message: String) {
+        NSLog(message)
```

Comment:
I would use `OSLog` with a `TailscaleKit` category / library specified, so that developers adopting this can filter logs coming from their app vs our code

Reply:
I want to avoid dragging in the os.log dependency.   The idea is that you can just inject your own logger - whatever that may be.  The long term goal is to get this to compile for non-apple platforms.